lib/Importer/Caffe2.cpp

```diff
@@ -270,6 +270,22 @@ void caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
 
     Node *node = G_.createConcat(opName, inputs, channel);
 
+    unsigned_t addAxis = dict.count("add_axis") ? loadInt(dict["add_axis"]) : 0;
+
+    if (addAxis) {
+      // When add axis is used, this means we have to add a new dimension before
+      // the axis, instead of merging on the axis.
+      std::vector<size_t> outputDims = inputs[0].dims();
+      for (const auto &input : inputs) {
+        GLOW_ASSERT(input.dims().size() > channel &&
+                    "input should have at least #axis dims");
+        GLOW_ASSERT(
```

Comment:
Admittedly I don't have a lot of experience in using our loader, but the thing that struck me is how little information it gives in release mode. Thus, I found those valuable in release mode as well.

I quickly had to use the debug version to realize that the input model had silly mistakes like this one.

What's our plan regarding error reporting?

In particular, I would rather have only sane networks hit the compiler. I guess we could run unconditionally the verifier in the compiler after loading the network, though today that wouldn't help since every checks in the verifier are done in asserts.

What do you think?

Reply:
This is orthogonal to the PR of course. LGTM on the PR.

What if you had a C++ program, accessing `std::vector` with incorrect indices, or something like that? There's a good chance that it would silently finish with wrong result.

I agree that yes, we need better error reporting. Right now asserts produce very short messages, don't inline numbers, can't print instruction number, etc. Often times after hitting assert I run `gdb` to see what exactly is going on.

In the other hand, I can imagine compiler being ran in production system, where performance is critical.
I understand that this particular check is lightweight, but when a lot of such checks add up, they may consume valuable CPU time.
I don't support running verifier in Release. What if user compiles the same net over and over again? They are absolutely sure that it's correct, and want to squeeze every single CPU cycle.

Probably this whole topic needs a discussion outside of PR :)